### ISSUE TITLE
Support in Vcpkg

### ISSUE THREAD

**Issue by luodaoyi** [mruby/mruby#6016]
Can mruby support in [vcpkg](https://github.com/microsoft/vcpkg)?

**Comment by rubyFeedback** [comment]
It may help laying down which steps would be necessary for that. What steps would be necessary for mruby to be available via vcpkg as well?

**Comment by luodaoyi** [comment]
[https://learn.microsoft.com/zh-cn/vcpkg/examples/packaging-github-repo](https://learn.microsoft.com/zh-cn/vcpkg/examples/packaging-github-repos)
[https://learn.microsoft.com/zh-cn/vcpkg/examples/packaging-zipfiles](https://learn.microsoft.com/zh-cn/vcpkg/examples/packaging-zipfiles)

like this ?

> It may help laying down which steps would be necessary for that. What steps would be necessary for mruby to be available via vcpkg as well?

**Comment by matz** [comment]
Thank you for the suggestion. Unfortunately, I don't know anything about vpkg. Actually, I have never programmed on Windows. If you (or anyone) send me a pull-request, I'd love to merge.

**Comment by luodaoyi** [comment]
It's use cmake ...

**Comment by dearblue** [comment]
Please note that I have never used vcpkg or written a cmake file, so this is all speculation.

As far as I can see, the vcpkg repository receives a pull request for build instructions for an arbitrary package.
The cmake file would be placed on the vcpkg side.
:crying_cat_face: No `ports/ruby`, so here's a link to python: https://github.com/microsoft/vcpkg/blob/2023.06.20/ports/python3/portfile.cmake

Also, it seems that cmake has a function to call external commands.
Therefore, it may be possible to use `rake DESTDIR=... install` from the vcpkg side.

In addition, it seems that vcpkg-specific patches for mruby can also be placed on the vcpkg side.

To sum up:

  - Probably no work needed in the mruby repository.
  - Probably someone interested will send a pull request to vcpkg (probably as `ports/mruby`).
  - Probably interested parties will update `ports/mruby` on the vcpkg side with a pull request.

**Comment by luodaoyi** [comment]
>
> Please note that I have never used vcpkg or written a cmake file, so this is all speculation.
> 
> As far as I can see, the vcpkg repository receives a pull request for build instructions for an arbitrary package. The cmake file would be placed on the vcpkg side. 😿 No `ports/ruby`, so here's a link to python: https://github.com/microsoft/vcpkg/blob/2023.06.20/ports/python3/portfile.cmake
> 
> Also, it seems that cmake has a function to call external commands. Therefore, it may be possible to use `rake DESTDIR=... install` from the vcpkg side.
> 
> In addition, it seems that vcpkg-specific patches for mruby can also be placed on the vcpkg side.
> 
> To sum up:
> 
> * Probably no work needed in the mruby repository.
> * Probably someone interested will send a pull request to vcpkg (probably as `ports/mruby`).
> * Probably interested parties will update `ports/mruby` on the vcpkg side with a pull request.

Correct!

**Comment by matz** [comment]
Closing — as @dearblue noted in 2023-06: vcpkg ports live in the microsoft/vcpkg repository, so there's no work needed here. The attempt at microsoft/vcpkg#32847 (by @diamant3) was closed without merge in 2023-09, and there's no `ports/mruby` in vcpkg today, but that's a vcpkg-side conversation rather than an mruby-side one.

mruby is already packaged via Homebrew, conda, ruby-install, ruby-build, and rvm (see the README), so users on most platforms have options. If someone in the community wants to drive a vcpkg port, the right venue is a PR to microsoft/vcpkg directly. Thanks all.
